### DISCUSSION TITLE
Map does not reflect setPaintProperty immediately

### DISCUSSION THREAD

**Discussion by usaken** [maplibre/maplibre-gl-js#2318] [Q&A] [answered]
With terrain, map does not reflect setPaintProperty immediately.
After setPaintProperty, map reflects it gradually by moving map.
Without terrain, map reflects setPaintProperty immediately.

I prepared the sample.
https://codepen.io/UsagiLabo/pen/QWVJddE

Clicking "test button" placed at top left of map sets raster-opacity to 0.1 usign setPaintProperty;
Should I have to add more scripts to update map immediately ? or is it bug of MapLibre ?
Is this same problem as below ?
https://github.com/maplibre/maplibre-gl-js/discussions/1782

Sorry for my poor English.

**Comment by HarelM** [comment, marked as answer]
There is an issue with opacity, I would advise to see if it happens with something else.

**Reply by usaken** [reply]
Thank you for your kind advice.

**Reply by kyllerss** [reply]
I've also run into this issue. This doesn't seem to me to be related to opacity (perhaps it is in a way I'm not observing). I have set the entire path to be black and I am still observing this behavior. 

The way I am currently getting around this issue, is by calling ```map.redraw()```, but this is very inefficient for my purposes (animating a path that has a gradient color based on speed - updating the color gradient stops on each iteration to create this animation effect). 

Any other workarounds or should I file a bug?
